packages/slate-editor/src/extensions/press-contacts/components/PressContactElement.tsx

```diff
@@ -1,32 +1,72 @@
-import type { ContactNode } from '@prezly/slate-types';
-import type { ContactInfo } from '@prezly/slate-types';
+import type { ContactInfo, ContactNode } from '@prezly/slate-types';
+import { ContactLayout } from '@prezly/slate-types';
 import classNames from 'classnames';
-import type { ReactNode } from 'react';
+import { useCallback } from 'react';
 import React from 'react';
+import type { FunctionComponent, ReactNode, SVGProps } from 'react';
 import type { RenderElementProps } from 'slate-react';
+import { useSlateStatic } from 'slate-react';
 
 import { Avatar, EditorBlock } from '#components';
-import { User } from '#icons';
+import { Envelope, Globe, Mobile, Phone, SocialFacebook, SocialTwitter, User } from '#icons';
+
+import { removePressContact, updatePressContact } from '../lib';
 
 import styles from './PressContactElement.module.scss';
+import { PressContactMenu } from './PressContactMenu';
 
 interface Props extends RenderElementProps {
     element: ContactNode;
     renderMenu?: (props: { onClose: () => void }) => ReactNode;
 }
 
 export function PressContactElement({ attributes, children, element, renderMenu }: Props) {
+    const editor = useSlateStatic();
+    const { layout, show_avatar: showAvatar } = element;
+    const isCardLayout = layout === ContactLayout.CARD;
+    const isSignatureLayout = layout === ContactLayout.SIGNATURE;
+
+    const handleToggleAvatar = useCallback(
+        (showAvatar: boolean) => updatePressContact(editor, element, { show_avatar: showAvatar }),
+        [editor, element],
+    );
+    const handleChangeLayout = useCallback(
+        (layout: ContactLayout) => updatePressContact(editor, element, { layout }),
+        [editor, element],
+    );
+    const handleRemove = useCallback(() => removePressContact(editor), [editor, element]);
+
     return (
         <EditorBlock
             {...attributes}
-            border
+            border={isCardLayout}
             element={element}
             // We have to render children or Slate will fail when trying to find the node.
             renderAboveFrame={children}
-            renderMenu={renderMenu}
+            renderMenu={(props) => {
+                if (renderMenu) {
+                    return renderMenu(props);
+                }
+
+                return (
+                    <PressContactMenu
```

Comment:
I need to add this link but the URL needs to be provided via plugin props. Just wasn't sure what would be a good name/structure for this type of thing? Any ideas?

I was thinking just going with something like "menuLinks: {url, label}[]" 🤔 

![Screenshot 2023-05-22 at 10 55 20](https://github.com/prezly/slate/assets/4209081/cabd3d7c-7463-44d0-8619-58ad68281cc0)

Reply:
> I was thinking just going with something like `menuLinks: {url, label}[]`

IDK, maybe.

FWIW when we needed to generate external links in other places, we've been passing _generate_ functions as extension parameters: 

https://github.com/prezly/slate/blob/e8ba684a9faa4a6a5304856bac7681f3a75c9ba6/packages/slate-editor/src/extensions/story-bookmark/types.ts#L14-L16